test/component/Tooltip/Tooltip.sync.spec.tsx

```diff
@@ -0,0 +1,382 @@
+//   name: 'FunnelChart',
+//   Wrapper: ({ children, syncId }) => (
+//     <FunnelChart width={700} height={500} syncId={syncId}>
+//       <Funnel isAnimationActive={false} dataKey="uv" nameKey="name" data={PageData} />
+//       {children}
+//     </FunnelChart>
+//   ),
+//   mouseHoverSelector: funnelChartMouseHoverTooltipSelector,
+// };
+
+// TODO: support synchronization in Sankey, Sunburst, Treemap
+// const SankeyTestCase: TooltipVisibilityTestCase = {
+//   name: 'Sankey',
+//   Wrapper: ({ children }) => (
+//     <Sankey width={400} height={400} margin={{ top: 20, right: 20, bottom: 20, left: 20 }} data={SankeyData}>
+//       {children}
+//     </Sankey>
+//   ),
+//   mouseHoverSelector: sankeyNodeChartMouseHoverTooltipSelector,
+//   expectedTransform: 'transform: translate(35px, 114.89236115144739px);',
+// };
+
+// const SunburstChartTestCase: TooltipVisibilityTestCase = {
+//   name: 'SunburstChart',
+//   Wrapper: ({ children }) => (
+//     <SunburstChart width={400} height={400} data={exampleSunburstData}>
+//       {children}
+//     </SunburstChart>
+//   ),
+//   mouseHoverSelector: sunburstChartMouseHoverTooltipSelector,
+//   expectedTransform: 'transform: translate(285px, 210px);',
+// };
+
+// const TreemapTestCase: TooltipVisibilityTestCase = {
+//   name: 'Treemap',
+//   Wrapper: ({ children }) => (
+//     <Treemap
+//       width={400}
+//       height={400}
+//       data={exampleTreemapData}
+//       isAnimationActive={false}
+//       nameKey="name"
+//       dataKey="value"
+//     >
+//       {children}
+//     </Treemap>
+//   ),
+//   mouseHoverSelector: treemapNodeChartMouseHoverTooltipSelector,
+//   expectedTransform: 'transform: translate(94.5px, 58.5px);',
+// };
+
+const testCases: ReadonlyArray<TooltipVisibilityTestCase> = [
+  AreaChartTestCase,
+  BarChartTestCase,
+  LineChartHorizontalTestCase,
+  LineChartVerticalTestCase,
+  ComposedChartWithAreaTestCase,
+  ComposedChartWithBarTestCase,
+  ComposedChartWithLineTestCase,
+  // FunnelChartTestCase,
+  // PieChartTestCase,
+  RadarChartTestCase,
+  RadialBarChartTestCase,
+  // ScatterChartTestCase,
+];
+
+// Tooltip sync does not work for PieChart, ScatterChart, FunnelChart, SunburstChart, SankeyChart, Treemap
+describe('Tooltip synchronization', () => {
+  describe.each(testCases)('as a child of $name', ({ name, Wrapper, mouseHoverSelector }) => {
+    test(`${name} shows tooltip when synchronized with ${name}`, () => {
+      const { container, debug } = render(
+        <>
+          <Wrapper syncId="tooltipSync">
+            <Tooltip />
+          </Wrapper>
+          <Wrapper syncId="tooltipSync">
+            <Tooltip />
+          </Wrapper>
+        </>,
+      );
+      const wrappers = container.querySelectorAll('.recharts-wrapper');
+      const wrapperTwo = wrappers[1];
+
+      // target the first chart to show the tooltip
+      showTooltip(container, mouseHoverSelector, debug);
```

Comment:
How do we know this targets the first chart? Should it instead do `showTooltip[wrappers[0], ...`?

Reply:
Yeah probably